.github/workflows/docker-images.yaml

```diff
@@ -22,19 +22,29 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        flavor:
-          - ""
-          - "-conda"
-          - "-kubernetes"
+        image:
+          - "prefecthq/prefect"
+          - "prefecthq/prefect-client"
+        release:
+          - true
+          - false
         python-version:
           - "3.9"
           - "3.10"
           - "3.11"
           - "3.12"
+        flavor:
+        - ""
+        include:
+        # Add conda and kubernetes flavors only for prefect images
+        - image: "prefect"
+          flavor: "-conda"
+        - image: "prefect"
+          flavor: "-kubernetes"
```

Comment:
@desertaxle just wanted to confirm my understanding matches yours.  If you add some extra items like this in `include:` and you _don't_ provide a key (like `python-version`), that will get exploded out according to the matrix, right?  So we should get both the pre-release and release versions of these across the 4 python versions?

Reply:
Looking at the [GitHub Actions docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#expanding-or-adding-matrix-configurations), I'm unsure if this will work or not. I think this might be a good case to use [`exclude`](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#excluding-matrix-configurations) instead of include because it would be more explicit. I think something like this would work:
```yaml
exclude:
    - image: "prefecthq/prefect-client"
      flavor: "-conda"
    - image: "prefecthq/prefect-client"
      flavor: "-kubernetes"
```
as long as all flavor entries are returned to their previous state.